sklearn/tree/_export.py

```diff
@@ -742,6 +742,26 @@ def recurse(self, node, tree, ax, max_x, max_y, depth=0):
             ax.annotate("\n  (...)  \n", xy_parent, xy, **kwargs)
 
 
+@validate_params(
+    {
+        "decision_tree": [DecisionTreeClassifier, DecisionTreeRegressor],
```

Comment:
I think we should be more lenient here. We could think of people implementing tree and exposing the same API and it should work.

Some how, we would expect `decision_tree` to have a `tree_` attribute.

Reply:
We can either use `HasMethods` even if it will be run for an attribute or not run any validation.

Reply:
I think `HasMethods` cannot be run for an attribute since it would check whether `hasattr` returns a callable. With that in mind, I may decided to not run any validation on `decision_tree`.

Reply:
Indeed, we have a `callable`. We should not validate then.